tests/docker_scripts/stateless_runner.sh

```diff
@@ -405,6 +413,23 @@ fi
 # stop logs replication to make it possible to dump logs tables via clickhouse-local
 stop_logs_replication
 
+# build jemalloc flamegraph if memory-related exceptions were returned from tests
+# todo add a step to grep the test output for memory related errors
+if [[ "$BUILD_JEMALLOC_REPORT" == "true" ]]; then
+  clickhouse-client --query "set send_logs_level='trace';system jemalloc flush profile;"
+  ls /tmp/jemalloc_clickhouse*
+  cp /tmp/jemalloc_clickhouse* /test_output
+
+  curl https://raw.githubusercontent.com/jemalloc/jemalloc/dev/bin/jeprof.in | sed s/@jemalloc_version@/5.0.1/ > jeprof
+  chmod +x jeprof
+
+  curl -LO https://github.com/brendangregg/FlameGraph/archive/refs/heads/master.zip && unzip -q master.zip && mv FlameGraph-master FlameGraph && rm master.zip
+  chmod +x ./FlameGraph/flamegraph.pl
+
+  ./jeprof "$(which clickhouse)" /tmp/jemalloc_clickhouse*.heap --collapsed  > output1.collapsed
+  ./FlameGraph/flamegraph.pl output1.collapsed --color mem --width 2560 > /test_output/jemalloc_flamegraph.svg
+fi
```

Comment:
I was actually thinking about dumping it on each `MEMORY_LIMIT_EXCEEDED` i.e. move this code into `clickhouse-test`, but I think we can start with this. Thanks!